deliver/lib/deliver/upload_metadata.rb

```diff
@@ -451,10 +459,43 @@ def retry_if_nil(message, tries: 5, wait_time: 10)
       end
     end
 
-    # Finding languages to enable
-    def verify_available_info_languages!(options, app, languages)
-      app_info = fetch_edit_app_info(app)
+    # Checking if the metadata to update includes localised App Info
+    def updating_localised_app_info?(options, app, app_info)
+      app_info ||= fetch_live_app_info(app)
+      unless app_info
+        UI.important("Can't find edit or live App info. Skipping upload.")
+        return false
+      end
+      localizations = app_info.get_app_info_localizations
 
+      LOCALISED_APP_VALUES.keys.each do |key|
+        current = options[key]
+        next unless current
+
+        unless current.kind_of?(Hash)
+          UI.error("Error with provided '#{key}'. Must be a hash, the key being the language.")
+          next
+        end
+
+        current.each do |language, value|
+          strip_value = value.to_s.strip
+          next if strip_value.empty?
+
+          app_info_locale = localizations.find { |l| l.locale == language }
+          next if app_info_locale.nil?
+
+          current_value = app_info_locale.public_send(key.to_sym)
+
+          return true if current_value != strip_value
+        end
+      end
+
+      UI.message('No changes to localised App Info detected. Skipping upload.')
+      return false
+    end
+
+    # Finding languages to enable
+    def verify_available_info_languages!(options, app, app_info, languages)
       unless app_info
         UI.user_error!("Cannot update languages - could not find an editable info")
```

Comment:
A bit outside of the scope of this PR, but while we're updating the code around that logic, I feel we could use message that could be a tad more helpful. Wdyt?

```suggestion
        UI.user_error!("Cannot update languages - could not find an editable version. Verify that your app is in one of the editable state in App Store Connect")
```

Reply:
Good point! 
I wonder if "version" would be confusing, as in this case the user may have an editable version (on one platform), but not editable App info (e.g. because the other platform is under review).
I used your wording, but switched 'version' to 'App info'.
Let me know if it doesn't look good to you.

Reply:
I'd use either quoted an capitalized `'App Info'` or simple nouns (thus all lowercase) `app info` @loremattei 😊

Reply:
I've pushed a commit to change to `'App Info'` so we can move forward with this PR, I hope you don't mind 😊

Reply:
One last thing! What do you guys think about the message I suggested [earlier](https://github.com/fastlane/fastlane/pull/21472#pullrequestreview-1599443550)? User-friendlier and more specific?

What I suggested earlier:

>Unable to upload metadata: It seems that the app you're attempting to upload metadata for isn't currently in an editable state across any platform. Please ensure that the app you're working with falls into one of these states: Prepare for Submission, Developer Rejected, Rejected, Metadata Rejected, Waiting for Review, or Invalid Binary.

What I think we could use now, after the latest changes to the logic:

>Unable to upload metadata: It seems that the app you're attempting to upload metadata for isn't currently in an editable state. Please ensure that the app you're working with falls into one of these states: Prepare for Submission, Developer Rejected, Rejected, Metadata Rejected, Waiting for Review, or Invalid Binary.

Reply:
One last thing! What do you guys think about the message I suggested [earlier](https://github.com/fastlane/fastlane/pull/21472#pullrequestreview-1599443550)? User-friendlier and more specific?

What I suggested earlier:

>Unable to upload metadata: It seems that the app you're attempting to upload metadata for isn't currently in an editable state across any platform. Please ensure that the app you're working with falls into one of these states: Prepare for Submission, Developer Rejected, Rejected, Metadata Rejected, Waiting for Review, or Invalid Binary.

What I think we could use now, after the latest changes to the logic:

>Unable to upload metadata: It seems that the app you're attempting to upload metadata for isn't currently in an editable state. Please ensure that the app you're working with falls into one of these states: Prepare for Submission, Developer Rejected, Rejected, Metadata Rejected, Waiting for Review, or Invalid Binary.

@loremattei @AliSoftware

Reply:
WFM 👍